provider/blob/src/export/blob_exporter.rs

```diff
@@ -6,74 +6,71 @@ use crate::blob_schema::*;
 use icu_provider::export::DataExporter;
 use icu_provider::prelude::*;
 use icu_provider::serde::SerializeMarker;
-use litemap::LiteMap;
+use std::sync::Mutex;
 use writeable::Writeable;
 use zerovec::map2d::ZeroMap2d;
 
 /// A data exporter that writes data to a single-file blob.
 /// See the module-level docs for an example.
 pub struct BlobExporter<'w> {
-    resources: LiteMap<(ResourceKeyHash, String), Vec<u8>>,
+    resources: Mutex<Vec<(ResourceKeyHash, String, Vec<u8>)>>,
     sink: Box<dyn std::io::Write + 'w>,
 }
 
+// This isn't auto-sync because sink is not Sync. But that's fine, because
+// it's only ever used in a &mut method, so there will be no aliases on any
+// thread.
+unsafe impl Sync for BlobExporter<'_> {}
```

Comment:
Issue: If people are forced to write `unsafe impl` when implementing DataExporter, it means there's a problem with DataExporter.  `unsafe impl` can be very unsafe and is hard to debug.  Since the `sink` field is only used in the `close()` method, maybe we need to pass more context into that method.  Or perhaps the `sink` field needs to be changed in a way that makes it threadsafe; can you make it wrap a `dyn Write + Sync` maybe?

Reply:
Big +1 here, we should really not be using unsafe impls here. Let's wrap the sink field.

You basically should never be doing `unsafe impl` for `Sync` or `Send` unless you're implementing something low level or doing some novel strategy. We're just using mutexes and such here, it should just work.

Reply:
Alternatively, we can have the `&mut self` method that writes to the sink take the sink as a parameter so that the exporter does not hold on to the actual sink.

Reply:
The close method cannot accept the sink because it's a blob specific argument. I didn't want to add the `Sync` restriction on sink because it's not needed. I've since discovered `Mutex::get_mut` which doesn't lock because it has a mut ref, so I'll just do that.